cmd/jaeger/internal/integration/storagecleaner/extension.go

```diff
@@ -52,7 +52,7 @@ func (c *storageCleaner) Start(ctx context.Context, host component.Host) error {
 		if !ok {
 			return fmt.Errorf("storage %s does not implement Purger interface", c.config.TraceStorage)
 		}
-		if err := purger.Purge(); err != nil {
+		if err := purger.Purge(ctx); err != nil {
```

Comment:
this is incorrect, you're using context from startup to execute purge in response to a request. At L62 the correct context to use is `r.Context()`. You should make this lambda function accept ctx arg.

Reply:
ok